app/sql.js

```diff
@@ -770,6 +772,85 @@ async function updateSchema(instance) {
     // eslint-disable-next-line no-await-in-loop
     await runSchemaUpdate(schemaVersion, instance);
   }
+  await updateLokiSchema(instance);
+}
+
+const LOKI_SCHEMA_VERSIONS = [
+  updateToLokiSchemaVersion2,
+];
+
+async function updateToLokiSchemaVersion2(currentVersion, instance) {
+  if (currentVersion >= 2) {
+    return;
+  }
+  console.log('updateToLokiSchemaVersion2: starting...');
+  await instance.run('BEGIN TRANSACTION;');
+
+  await instance.run(
+    `CREATE TABLE pairingAuthorisations(
+      id INTEGER NOT NULL PRIMARY KEY AUTOINCREMENT,
+      issuerPubKey VARCHAR(255),
+      secondaryDevicePubKey VARCHAR(255),
+      signature VARCHAR(255),
+      json TEXT
+    );`
+  );
+
+  await instance.run(
+    `INSERT INTO loki_schema (
+        version
+      ) values (
+        2
+      );`
+  );
+  await instance.run('COMMIT TRANSACTION;');
+  console.log('updateToLokiSchemaVersion2: success!');
+}
+
+async function updateLokiSchema(instance) {
+  const result = await instance.get("SELECT name FROM sqlite_master WHERE type = 'table' AND name='loki_schema'");
+  if (!result) {
+    await createLokiSchemaTable(instance);
+  }
+  const lokiSchemaVersion = await getLokiSchemaVersion(instance);
+  console.log(
+    'updateLokiSchema:',
+    `Current loki schema version: ${lokiSchemaVersion};`,
+    `Most recent schema version: ${LOKI_SCHEMA_VERSIONS.length};`
+  );
+  for (let index = 0, max = LOKI_SCHEMA_VERSIONS.length; index < max; index += 1) {
```

Comment:
why not use `LOKI_SCHEMA_VERSIONS.forEach(...)`?

Reply:
This snippet is currently living in the public channel branch, so to reduce conflicts I just copy/pasted it here. But once merged, we could definitely revisit it!

Reply:
The reason they are using a raw for loop here is that forEach executes each iteration asynchronously, i.e. after the first await in the first iteration it could start executing the next
We explicitly want to complete each iteration atomically before starting the next